lib/osf-components/addon/components/contributors/add-unregistered-modal/component.ts

```diff
@@ -0,0 +1,60 @@
+import { action } from '@ember/object';
+import Component from '@glimmer/component';
+import { ValidationObject } from 'ember-changeset-validations';
+import { validateFormat, validatePresence } from 'ember-changeset-validations/validators';
+import { Permission } from 'ember-osf-web/models/osf-model';
+import buildChangeset from 'ember-osf-web/utils/build-changeset';
+
+interface AddUnregisteredModalArgs {
+    addUnregisteredContributor: (
+        email: string,
+        fullName: string,
+        permission: Permission,
+        bibliographic: boolean,
+    ) => null;
+}
+
+interface UnregisteredFormFields {
+    email: string;
+    fullName: string;
+    permission: string;
+    bibliographic: boolean;
+}
+
+const UnregisteredFormValidations: ValidationObject<UnregisteredFormFields> = {
+    email: [
+        validateFormat({
+            type: 'email',
+            allowBlank: true,
+            translationArgs: { description: 'This field' },
+        }),
+    ],
+    fullName: validatePresence({
+        presence: true,
+        type: 'empty',
+    }),
+};
+
+export default class UserSearchComponent extends Component<AddUnregisteredModalArgs> {
+    unregisteredFormChangeset = buildChangeset({
+        email: null,
+        fullName: null,
+        permission: Permission.Write,
+        bibliographic: true,
+    }, UnregisteredFormValidations);
+    permissionOptions = [...Object.values(Permission)];
+
+    @action
+    validate() {
+        this.unregisteredFormChangeset.validate();
```

Comment:
Does `changeset.validate()` already return a promise? and the queue helper is able to wait on it to resolve or reject.

Reply:
`changeset.validate()` does not reject even if there is a validation error. Therefore we have to wrap it in this action and return a rejected promise to stop execution of subsequent functions passed to the `queue` helper.